src/test/java/com/intuit/fuzzymatcher/component/MatchServicePerfTest.java

```diff
@@ -81,4 +91,39 @@ static String getAddress(String[] csv) {
         }
         return address;
     }
+
+    private static CSVReader getCSVReader(String FileName) throws FileNotFoundException {
+        return new CSVReaderBuilder(
+                new FileReader(MatchServicePerfTest.class.getClassLoader().getResource(FileName).getFile()))
+                .withSkipLines(1)
+                .build();
+    }
+
+    private List<Document> readTxnFromCSV(String fileName, AtomicInteger index) throws FileNotFoundException {
+
+        List<Document> list = new ArrayList<>();
+        StreamSupport.stream(getCSVReader(fileName).spliterator(), false).forEach(csv -> {
+            int i =0;
+            list.add(new Document.Builder(index.getAndIncrement()+"")
+                    .addElement(new Element.Builder().setValue(csv[0]).setType(TEXT).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[1]).setType(TEXT).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[2]).setType(TEXT).setVariance(i++ + "").createElement())// should be date
+                    .addElement(new Element.Builder().setValue(csv[3]).setType(NUMBER).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[4]).setType(NUMBER).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[5]).setType(NUMBER).setVariance(i++ + "").createElement())
```

Comment:
Adding threshold to number and date type elements will reduce the memory requirement. The Default threshold at element level is 0.3 . For Number and Date type they do not give relevant matches
For example the Number value of 30 will match with 100. For these 2 types a threshold of close to 0.9 will give more relevant results
Try replacing it with this
```
                    .addElement(new Element.Builder().setValue(csv[2]).setType(NUMBER).setVariance(i++ + "").setThreshold(1.0).createElement())// should be date
                    .addElement(new Element.Builder().setValue(csv[3]).setType(NUMBER).setVariance(i++ + "").setThreshold(0.9).createElement())
                    .addElement(new Element.Builder().setValue(csv[4]).setType(NUMBER).setVariance(i++ + "").setThreshold(0.9).createElement())
                    .addElement(new Element.Builder().setValue(csv[5]).setType(NUMBER).setVariance(i++ + "").setThreshold(0.9).createElement())
```
You will see a reduction of over 2GB in runtime memory for the dataset